cometd-java/cometd-java-oort/src/main/java/org/cometd/oort/Seti.java

```diff
@@ -385,6 +385,42 @@ public boolean disassociate(final String userId, ServerSession session)
         return removed;
     }
 
+    /**
+     * <p>Disassociates the given userId from all sessions.</p>
+     * <p>If this user had sessions on this cometd, broadcasts this information
+     * on the Oort cloud, so that other comets will know that the given userId no longer is on this comet.</p>
+     *
+     * @param userId the user identifier to disassociate
+     * @return set of sessions that were disassociated
+     * @see #associate(String, ServerSession)
+     */
+    public Set<ServerSession> disassociate(final String userId)
+    {
+        final Set<LocalLocation> userLocations = new HashSet<>();
+        synchronized(_uid2Location) {
+            for(Location location : _uid2Location.get(userId))
+            {
+                if (location instanceof LocalLocation)
+                {
+                    userLocations.add((LocalLocation) location);
+                }
+            }
+        }
+
+        final Set<ServerSession> removedUserSessions = new HashSet<>();
+        for(LocalLocation location : userLocations)
+        {
+            final ServerSession session = location._session;
+            boolean removed = disassociate(userId, session);
+            if(removed)
+            {
+                removedUserSessions.add(session);
+            }
+        }
+
+        return removedUserSessions;
+    }
+
```

Comment:
Please stick with the spacing and brace positioning of the rest of the code. For example, space after `for`, no space after cast, brace on new line, etc.

Your implementation is "expensive", in that allocates 2 `Set`s, calls `disassociate(String, ServerSession)` which allocates another `LocalLocation` that you already have, etc.
However, I tried to write another version and it was much more complicated and required more refactorings, so I prefer yours. 

Just rebase it on top of branch `3.0.x` and I will merge it.

Thanks !